File: sorting/java/selection_sort.java
import java.util.*;

class selection_sort {
    public Vector<Integer> Selection_sort(Vector<Integer> list){
        int n = list.size();
        for (int p=0; p<n; p++){
            int Min = p;
            for (int i=p; i<n; i++){
                if(list.get(i)<list.get(Min)){//find the index of smallest element
                    Min = i;
                }
            }
            int temp = list.get(p);     //swapping the min to end of sorted part
            list.set(p, list.get(Min));
            list.set(Min, temp);
        }
        return list;
    }
}

File: sorting/python/selection_sort.py
def selection_sort(List):
	n = len(List)
	for pointer in range(n):
		Min = pointer
		for i in range(pointer, n):	#find the index of smallest element
			if(List[i] < List[Min]):
				Min = i
		temp = List[pointer]	#swap
		List[pointer] = List[Min]
		List[Min] = temp
	return List